cmake/therock_subproject.cmake

```diff
@@ -945,10 +958,24 @@ function(_therock_cmake_subproject_setup_toolchain
   string(APPEND _toolchain_contents "set(CMAKE_C_COMPILER_LAUNCHER \"@CMAKE_C_COMPILER_LAUNCHER@\")\n")
   string(APPEND _toolchain_contents "set(CMAKE_CXX_COMPILER_LAUNCHER \"@CMAKE_CXX_COMPILER_LAUNCHER@\")\n")
   string(APPEND _toolchain_contents "set(CMAKE_MSVC_DEBUG_INFORMATION_FORMAT \"@CMAKE_MSVC_DEBUG_INFORMATION_FORMAT@\")\n")
-  string(APPEND _toolchain_contents "set(CMAKE_C_FLAGS_INIT @CMAKE_C_FLAGS@)\n")
-  string(APPEND _toolchain_contents "set(CMAKE_CXX_FLAGS_INIT @CMAKE_CXX_FLAGS@)\n")
-  string(APPEND _toolchain_contents "set(CMAKE_EXE_LINKER_FLAGS_INIT @CMAKE_EXE_LINKER_FLAGS@)\n")
-  string(APPEND _toolchain_contents "set(CMAKE_SHARED_LINKER_FLAGS_INIT @CMAKE_SHARED_LINKER_FLAGS@)\n")
+  if(MSVC)
+    # The system compiler and the toolchain compiler are incompatible, so we
+    # define flags from scratch for the toolchain compiler. Each ROCm project
+    # typically has its own `toolchain-windows.cmake` file that we are bypassing
+    # here. If any flags are load bearing we can either add them to all projects
+    # or source those flags from the projects themselves more locally.
+    string(APPEND _toolchain_contents "set(CMAKE_C_FLAGS_INIT )\n")
+    string(APPEND _toolchain_contents "set(CMAKE_CXX_FLAGS_INIT \"-DWIN32 -D_CRT_SECURE_NO_WARNINGS\")\n")
```

Comment:
Looks like we'll at least also need `-DNOMINMAX` for some sub-projects:

```
[build] [rocFFT] D:/projects/TheRock/math-libs/rocFFT/library/src/tuning_kernel_tuner.cpp:142:25: error: expected unqualified-id
[build] [rocFFT]   142 |         min_size = std::min(factors.size(), min_size);
[build] [rocFFT]       |                         ^
[build] [rocFFT] C:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\shared\minwindef.h:197:29: note: expanded from macro 'min'
[build] [rocFFT]   197 | #define min(a,b)            (((a) < (b)) ? (a) : (b))
[build] [rocFFT]       |                             ^
```

Reply:
Yeah. I wish msft would have just added a `-DWIN32_BE_NORMAL=1`. All part of the usual pack of insanity reduction flags.